packages/utils/crates/utils-types/src/sma.rs

```diff
@@ -99,20 +117,20 @@ mod tests {
 
     #[test]
     fn test_nosum_sma_empty() {
-        let sma = NoSumSMA::<u32>::new(3);
+        let sma = NoSumSMA::<u32, u32>::new(3);
         assert_eq!(3, sma.window_size(), "invalid windows size");
-        assert_eq!(0, sma.get_average(), "invalid empty average");
+        assert!(sma.get_average().is_none(), "invalid empty average");
     }
 
     #[test]
     fn test_nosum_sma_should_calculate_avg_correctly() {
-        let mut sma = NoSumSMA::<u32>::new(3);
+        let mut sma = NoSumSMA::<u32, u32>::new(3);
         sma.add_sample(0);
         sma.add_sample(1);
         sma.add_sample(2);
         sma.add_sample(3);
 
-        assert_eq!(2, sma.get_average(), "invalid average");
+        assert_eq!(2, sma.get_average().unwrap(), "invalid average");
```

Comment:
```suggestion
        assert_eq!(Some(2), sma.get_average(), "invalid average");
```

Reply:
Why? This is the same thing :)

Reply:
But has a more expressive output on test failure.